modules/xfem/src/base/XFEMAppTypes.C

```diff
@@ -8,6 +8,4 @@
 //* https://www.gnu.org/licenses/lgpl-2.1.html
 
 #include "XFEMAppTypes.h"
-#include "ExecFlagRegistry.h"
-
-const ExecFlagType EXEC_XFEM_MARK = registerDefaultExecFlag("XFEM_MARK");
+#include "ExecFlagRegistry.h"
```

Comment:
mv this file to solid_mechanics/src/base and rename something like:  solid_mechanicsAppTypes.C and h

Only include the execution flag part of this file. 

In the original XFEMAppTypes.h file move 

typedef unsigned int CutSubdomainID 

to XFEMApp.h right after the includes.